### ISSUE TITLE
Add a `--fail-fast` and/or `--best-effort` flag to control when to quit when errors are encountered

### ISSUE THREAD

**Issue by ghostsquad** [pulumi/pulumi#9422]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

This is a feature request to make the "feature" described in #8005 a cli-flag. In some scenarios, I would like pulumi to try to do as much as it can, even when an error occurs with a particular resource.

### Affected area/feature

Pulumi CLI / Automation API

**Comment by Frassle** [comment]
Isn't this the opposite of #8005? That was asking for pulumi to exit faster as soon as it hit an error (which is unsafe because of pending operations). Sounds like this is asking for pulumi to keep going as much as it can when it hits an error? Which is more of a --no-fail-fast, or --persevere option?

**Comment by ghostsquad** [comment]
@Frassle yes I meant the original behavior before it was changed with #8005. Ok if the default remains the same, but if a flag allowed continued execution that would be great.

Maybe this is a bit of scope creep, but if there was a programmatic way to know that an error occurred, such that the program can be part of the decision making process of when to fail?

I believe the typical thing would be to use the standard context.Context (go), which could be made public (or use a getter) in the pulumi.Context, and we can watch that to know about cancellations and deadlines.

**Comment by Frassle** [comment]
We didn't change any behavior in #8005, it was just closed as answered. 

We can give this a think. There might be some difficulties with correctly resolving output promises but the general idea of a `--persevere` option sounds fine.

**Comment by ghostsquad** [comment]
Oh sorry, your last question made me second guess what I previously read in #8005.

Basically, if you have a dependency graph, specific branches of the tree will not be able to be completed because of failures. But there may be other parts of the graph that are totally unaffected.

It would be nice to be able to essentially run as much as possible, being the branches that are siblings to the failure.

**Comment by Frassle** [comment]
> It would be nice to be able to essentially run as much as possible, being the branches that are siblings to the failure.

Yup understood, I've called this `--best-effort` or `--persevere` in previous tools.
